File: div11/00.py
import time, os

start = time.monotonic()

count = 400000

lst = []
for i in range(102345678, 987654321):
    if len(set(str(i))) == len(str(i)):
        if i % 11 == 0:
            lst.append(i)
            if len(lst) == count:
                break

print("count =", count)
print("len(lst) =", len(lst))
print(lst[0], lst[-1])

finish = time.monotonic()
result = finish - start
print(round(result,2), "сек")

print("- далее просто первый и последний -")

for i in range(102345678, 987654321):
    if len(set(str(i))) == len(str(i)):
        if i % 11 == 0:
            print(i)
            break
for i in range(987654321, 102345678-1, -1):
    if len(set(str(i))) == len(str(i)):
        if i % 11 == 0:
            print(i)
            break

File: 00-Основы научных знаний/div11/00.js
console.time("time")

count = 400000

let lst = []
for (let i = 102345678; i <= 987654321; i++) {
    if ((new Set(String(i).split("")).size) == 9) {
        if (i % 11 === 0) {
            lst.push(i)
            if (lst.length === count)
                break    
        }
    }
}

console.log("count =", count)
console.log("len(lst) =", lst.length)
console.log(lst[0], lst[lst.length-1])

console.timeEnd("time")

console.log("- далее просто первый и последний -")

for (let i = 102345678; i <= 987654321; i++) {
    if ((new Set(String(i).split("")).size) == 9) {
        if (i % 11 === 0) {
            console.log(i)
            break
        }
    }
}

for (let i = 987654321; i >= 102345678; i--) {
    if ((new Set(String(i).split("")).size) == 9) {
        if (i % 11 === 0) {
            console.log(i)
            break
        }
    }
}
